### ISSUE TITLE
Syntax highlighting does not support negative literals

### ISSUE THREAD

**Issue by ueoo** [microsoft/pylance-release#6267]
I am using latest VS Code and Python/Pylance extension on macOS.
The color theme is the Dracula https://github.com/dracula/visual-studio-code
The minus symbol `-` is not highligted with the numbers.
<img width="249" alt="image" src="https://github.com/user-attachments/assets/3ba88917-b030-44fb-924e-c3584e3ac1a1">
I suggest to add something like this \-? at the begin of this regexp.
Like this: (\-?[0-9]+)
A similar issue of other extension: https://github.com/hashicorp/vscode-terraform/issues/243

**Comment by hologerry** [comment]
Hi, @karthiknadig  thanks! Hope we can have a solution to this.

**Comment by rchiodo** [comment]
I don't think this is an actual bug. You'd have to change the theme. 

That character is identified as keyword.operator.arithmetic



![Image](https://github.com/user-attachments/assets/67adfe4d-b9f3-4ec1-a8c8-1071ee079796)

The same as *



![Image](https://github.com/user-attachments/assets/9fce9087-333c-42e1-a108-e9709d29fa4c)

So `-` counts as an operator.

**Comment by debonte** [comment]
> So `-` counts as an operator.

Counts as an operator in what sense? I see that the TextMate grammar is treating it that way, but I don't think that's true from Python's perspective.

Here's what I see in the [Python grammar](https://docs.python.org/3.12/reference/grammar.html), which indicates that negative numbers are literals.

```text
literal_pattern:
    | signed_number !('+' | '-') 
    | complex_number 
    | strings 
    | 'None' 
    | 'True' 
    | 'False' 

...

signed_number:
    | NUMBER
    | '-' NUMBER
```

Then for fun I tried https://github.com/iritkatriel/codoscope and it also indicates that there's no operator call involved here:

![image](https://github.com/user-attachments/assets/f4bbe1b8-09a0-45d4-acfc-eb96dfef00f6)

**Comment by rchiodo** [comment]
Wouldn't the UNARY_NEGATIVE be the operator?

**Comment by debonte** [comment]
Yeah, I added the `y = -x` line to compare the two scenarios. Sorry that wasn't clear. The highlighted bytecode instructions on the right side are the entirety of the bytecode for `x = -1`.

**Comment by debonte** [comment]
Reopening since this is a bug. However, @hologerry, be aware that it may not get fixed anytime soon. See https://github.com/microsoft/pylance-release/issues/5824 for some context.

**Comment by hologerry** [comment]
Hi @debonte, Thanks for clarifying this! It’s great to know, and I hope we can get a better syntax highlighter in the future.

**Comment by heejaechang** [comment]
Repro workspace for investigation:
https://github.com/heejaechang/pylance-repro/tree/main/pylance-issues/6267

**Comment by heejaechang** [comment]
This issue has been fixed in prerelease version 2026.3.100, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20263100-24-August-2026-PreRelease)
